clients/bigquery/cast.go

```diff
@@ -104,3 +90,22 @@ func castColVal(colVal any, colKind columns.Column, additionalDateFmts []string)
 	slog.Error("Unexpected BigQuery Data Type", slog.Any("colKind", colKind.KindDetails.Kind), slog.Any("colVal", colVal))
 	return fmt.Sprint(colVal), nil
 }
+
+// EncodeStructToJSONString takes a struct as either a string or Go object and encodes it into a JSON string.
+// Structs from relational and Mongo are different.
+// MongoDB will return the native objects back such as `map[string]any{"hello": "world"}`
+// Relational will return a string representation of the struct such as `{"hello": "world"}`
+func EncodeStructToJSONString(value any) (string, error) {
+	if colValString, isOk := value.(string); isOk {
+		if strings.Contains(colValString, constants.ToastUnavailableValuePlaceholder) {
+			return fmt.Sprintf(`{"key":"%s"}`, constants.ToastUnavailableValuePlaceholder), nil
+		}
+		return colValString, nil
+	}
+
+	colValBytes, err := json.Marshal(value)
```

Comment:
@Tang8330 if `value` isn't a string do we care if the toast placeholder appears within it (e.g. `["__debezium_unavailable_value"]`, `{"foo": "__debezium_unavailable_value"}`)?

Reply:
I don't think it's possible, but can we add a guardrail for it and Sentry when it does happen?